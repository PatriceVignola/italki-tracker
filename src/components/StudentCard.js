/**
 * @prettier
 * @flow
 */

import React from 'react';
import {
  Button,
  Card,
  CardActions,
  CardContent,
  CardMedia,
  Paper,
  Typography,
} from 'material-ui';

import type {Student} from '../actions/types';

type Props = {
  student: Student,
};

function StudentCard(props: Props) {
  return (
    <Paper>
      <Card style={styles.card}>
        <CardMedia style={styles.cardMedia} image={props.student.photoUrl} />
        <CardContent>
          <Typography gutterBottom variant="headline" component="h2">
            {`${props.student.firstName} ${props.student.lastName}`}
          </Typography>
          <Typography component="p">
            Description here (e.g. level, number of classes, etc.)
          </Typography>
        </CardContent>
        <CardActions>
          <Button size="small" color="primary">
            Cancel
          </Button>
          <Button size="small" color="primary">
            OK
          </Button>
        </CardActions>
      </Card>
    </Paper>
  );
}

const styles = {
  cover: {
    height: '100%',
    resizeMode: 'cover',
  },
  card: {
    height: 500,
  },
  cardMedia: {
    height: 300,
  },
};

export default StudentCard;
