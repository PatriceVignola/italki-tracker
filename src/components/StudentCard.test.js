/**
 * @prettier
 * @flow
 */

import React from 'react';
import {shallow} from 'enzyme';

import StudentCard from './StudentCard';

describe('StudentList', () => {
  it('renders correctly', () => {
    const mockStudent = {
      id: 1,
      firstName: 'FirstName',
      lastName: 'LastName',
      photoUrl:
        'http://education.mnhs.org/immigration/sites/education.mnhs.org.immigration/files/imagecache/Full_800x800/MaleSilhouette_5.png',
    };

    const wrapper = shallow(<StudentCard student={mockStudent} />);
    expect(wrapper).toMatchSnapshot();
  });
});
