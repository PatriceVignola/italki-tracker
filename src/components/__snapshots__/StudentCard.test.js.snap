// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentList renders correctly 1`] = `
<WithStyles(Paper)>
  <WithStyles(Card)
    style={
      Object {
        "height": 500,
      }
    }
  >
    <WithStyles(CardMedia)
      image="http://education.mnhs.org/immigration/sites/education.mnhs.org.immigration/files/imagecache/Full_800x800/MaleSilhouette_5.png"
      style={
        Object {
          "height": 300,
        }
      }
    />
    <WithStyles(CardContent)>
      <WithStyles(Typography)
        component="h2"
        gutterBottom={true}
        variant="headline"
      >
        FirstName LastName
      </WithStyles(Typography)>
      <WithStyles(Typography)
        component="p"
      >
        Description here (e.g. level, number of classes, etc.)
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
    <WithStyles(CardActions)>
      <WithStyles(Button)
        color="primary"
        size="small"
      >
        Cancel
      </WithStyles(Button)>
      <WithStyles(Button)
        color="primary"
        size="small"
      >
        OK
      </WithStyles(Button)>
    </WithStyles(CardActions)>
  </WithStyles(Card)>
</WithStyles(Paper)>
`;
